src/qibo/quantum_info/quantum_networks.py

```diff
@@ -1,7 +1,8 @@
 """Module defining the `QuantumNetwork` class and adjacent functions."""
 
-import re
 from functools import reduce
+from logging import warning
+from numbers import Number
```

Comment:
Do we actually need this?

Reply:
@Canoming This question still remains: we don't need this `Number` class for typing, do we?